Modules/TRD/src/TrackingTask.cxx

```diff
@@ -0,0 +1,295 @@
+
+  for (const auto& tracktrig : tracktrigArr) {
+    int first = tracktrig.getFirstTrack();
+    int end = first + tracktrig.getNumberOfTracks();
+    // count number of matched tracks
+    int nMatched = 0;
+    for (int itrack = first; itrack < end; ++itrack) {
+      auto& trackqc = trackArr[itrack];
+      // type of track is 0 for TPC-TRD track, 1 for ITS-TPC-TRD track
+      if (trackqc.type == 0)
+        continue;
+      nMatched++;
+      // remove tracks with pt < 1 GeV/c
+      if (trackqc.pt < 1)
+        continue;
+      mNtracklets->Fill(trackqc.nTracklets);
+      mTrackPt->Fill(trackqc.pt);
+      mTrackChi2->Fill(trackqc.reducedChi2);
+      // flag to fill some objects only once per track
+      bool fillOnce = false;
+      for (auto ilayer : { 0, 1, 2, 3, 4, 5 }) {
+        // cut on x position of the track
+        if (trackqc.trackX[ilayer] < 2)
+          continue;
+        // calculate phi taking into account the rotation of the sector
+        double trackphi = TMath::ATan2(trackqc.trackY[ilayer], trackqc.trackX[ilayer]) + TMath::Nint(TMath::Floor(trackqc.trackletDet[ilayer] / 30)) * TMath::DegToRad() * 20 + TMath::DegToRad() * 10;
+        // find charge bin of the track
+        int charge = -1;
+        if (trackqc.trackQpt[ilayer] > 0)
+          charge = 0;
+        if (trackqc.trackQpt[ilayer] < 0)
+          charge = 1;
+        if (charge == -1)
+          continue;
+        if (!fillOnce) {
+          mTrackEta->Fill(trackqc.trackEta[ilayer]);
+          mTrackPhi->Fill(trackphi);
+          mTrackletsEtaPhi[charge]->Fill(trackqc.trackEta[ilayer], trackphi, trackqc.nTracklets);
+          fillOnce = true;
+        }
+        // fill eta-phi map of tracks
+        mTracksEtaPhiPerLayer[charge][ilayer]->Fill(trackqc.trackEta[ilayer], trackphi);
+        // residuals
+        mDeltaY->Fill(trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+        mDeltaZ->Fill(trackqc.trackZ[ilayer] - trackqc.trackletZ[ilayer]);
+        mDeltaYDet->Fill(trackqc.trackletDet[ilayer], trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+        mDeltaZDet->Fill(trackqc.trackletDet[ilayer], trackqc.trackZ[ilayer] - trackqc.trackletZ[ilayer]);
+        mDeltaYsphi->Fill(trackqc.trackSnp[ilayer], trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+        mDeltaYinEtaPerLayer[ilayer]->Fill(trackqc.trackEta[ilayer], trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+        mDeltaYinPhiPerLayer[ilayer]->Fill(trackphi, trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+        mTrackletDef->Fill(trackqc.trackSnp[ilayer], trackqc.trackletDy[ilayer]);
+        mDeltaYDet->Fill(trackqc.trackletDet[ilayer], trackqc.trackY[ilayer] - trackqc.trackletY[ilayer]);
+      } // end of loop over layers
+    }   // end of loop over tracks
+    mNtracks->Fill(nMatched);
+  } // end of loop over track trigger records
+}
+
+void TrackingTask::endOfCycle()
+{
+  ILOG(Info, Support) << "endOfCycle" << ENDM;
```

Comment:
Can you set all these messages level to `Debug` ?

Reply:
yeah, sure.